### PR TITLE
use user-data scheme for user data in shared process 

### PR DESCRIPTION
**Pull request by sandy081**
Re #126376 

This PR changes shared process to use user-data scheme for user data.

### CODE REVIEW

**Comment by sandy081** [comment]
@bpasero It seems change events are not working with these changes. For eg, try following:

- Enable settings sync 
- Turn on settings sync
- Open Settings Sync log
- Change a setting and see that settings sync is not triggered.

**Comment by bpasero** [comment]
@sandy081 to isolate the issue, I have added the following code to print file changes and to listen to changes to the global settings file:

```ts
console.log("starting to watch...");
fileService.watch(URI.file('/Users/bpasero/Desktop/settings-sync-ws/User/settings.json').with({ scheme: Schemas.userData }));
fileService.onDidFilesChange(e => {
	console.log(e);
});
```

I am seeing file change events printed when I open shared process. Can you check?

File: src/vs/platform/sharedProcess/node/sharedProcessEnvironmentService.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import { memoize } from 'vs/base/common/decorators';
+import { Schemas } from 'vs/base/common/network';
+import { URI } from 'vs/base/common/uri';
+import { NativeEnvironmentService } from 'vs/platform/environment/node/environmentService';
+
+export class SharedProcessEnvironmentService extends NativeEnvironmentService {
+
+	@memoize
+	override get userRoamingDataHome(): URI { return this.appSettingsHome.with({ scheme: Schemas.userData }); }
```
Comment (bpasero) [inline src/vs/platform/sharedProcess/node/sharedProcessEnvironmentService.ts:14]: Is this applying to all settings? I am not entirely user what "user roaming data" is?

File: src/vs/code/electron-browser/sharedProcess/sharedProcessMain.ts [hunk 2]
```diff
@@ -204,6 +207,10 @@ class SharedProcessMain extends Disposable {
 		const diskFileSystemProvider = this._register(new DiskFileSystemProvider(logService));
 		fileService.registerProvider(Schemas.file, diskFileSystemProvider);
 
+		const localFileSystemProviderFromMain = this._register(new IPCFileSystemProvider(mainProcessService.getChannel('localFilesystem'), { pathCaseSensitive: isLinux }));
```
Comment (bpasero) [inline src/vs/code/electron-browser/sharedProcess/sharedProcessMain.ts:210]: After this is merged I can see how to make this more reusable, i.e. have some factory method.

**Review by bpasero — APPROVED** [review]

File: src/vs/platform/sharedProcess/node/sharedProcessEnvironmentService.ts (hunk 1 above)
Comment (sandy081) [reply src/vs/platform/sharedProcess/node/sharedProcessEnvironmentService.ts:14]: yes - this is home for all user data - settings, keybindings, snippets. This is same as in workbench - 

https://github.com/microsoft/vscode/blob/33f84b36fd6757d37d3c8f5e2da0653b371641ec/src/vs/workbench/services/environment/electron-sandbox/environmentService.ts#L60

**Pull request merged**